great_expectations_cloud/agent/agent.py

```diff
@@ -173,7 +189,16 @@ def _listen(self) -> None:
             print("Received request to shut down.")
         except (SubscriberError, ClientError):
             print("The connection to GX Cloud has encountered an error.")
-        except (AuthenticationError, ProbableAuthenticationError, AMQPConnectorException):
+        except GXAgentUnrecoverableConnectionError:
+            print("The connection to GX Cloud has encountered an unrecoverable error.")
+            sys.exit(1)
```

Comment:
We catch the `GXAgentUnrecoverableConnectionError` at the top level and do the `sys.exit` here